### ISSUE TITLE
The generated image content and text do not match

### ISSUE THREAD

**Issue by ZZfive** [FrozenBurning/Text2Light#19]
Hey @FrozenBurning

Thanks for your awesome contribution! I am very interested in this job and have conducted some tests, and currently had some issues.

---

Using the provided weight file, there is a significant deviation between the image content and the text，such as：

# example 1
comand:  python text2light.py -rg logs/global_sampler_clip -rl logs/local_sampler --outdir ./generated_panorama --text "purple petal flower" --clip clip_emb.npy --sritmo ./logs/sritmo.pth --sr_factor 4

The images generated by two different inferences using the above command are significantly different from the text(**purple petal flower**)：
![image](https://github.com/FrozenBurning/Text2Light/assets/57706634/2c87ca8f-c536-4c34-a45a-44bb21adc66f)

![image](https://github.com/FrozenBurning/Text2Light/assets/57706634/025af0e9-1fa5-434a-95c8-3f37652a368d)

# example 2
comand:  python text2light.py -rg logs/global_sampler_clip -rl logs/local_sampler_outdoor --outdir ./generated_panorama --text "Elephant, Watering Hole, Baby Elephant" --clip clip_emb.npy --sritmo ./logs/sritmo.pth --sr_factor 4

![image](https://github.com/FrozenBurning/Text2Light/assets/57706634/fe758ed8-bd60-4e24-bc0e-1f85b4523752)

---

Except for the above，with using "**eval mode**" in the model and $@torch.no_grad()$ in inferencing, it seems that the previous inference will still affect the next inference result, causing the content of the image generated by the current inference to be similar to the previous text, with a significant deviation from the input text content of the current inference.

And regardless of the type of local sampler used, the generated hrldr often has darker and colder tones, but the hdr is also overexposed, such as

# example1
hrldr:
![image](https://github.com/FrozenBurning/Text2Light/assets/57706634/7b1e9fb9-8529-4666-a134-af46beef2c70)

hdr:
![image](https://github.com/FrozenBurning/Text2Light/assets/57706634/c3bf47a7-af83-4895-b0d3-0b699e734f41)

# example2
hrldr:
![image](https://github.com/FrozenBurning/Text2Light/assets/57706634/b6afa8c0-c506-4de9-8eca-b78a5c65bcd3)

hdr:
![image](https://github.com/FrozenBurning/Text2Light/assets/57706634/765d9762-0b77-4500-b891-104bb1ee90c2)

Which parameters can be adjusted to improve the generation quality of hrldr and hdr images？？？

---

Also, please tell me how to handle the above problem。

Thanks!!!

**Comment by FrozenBurning** [comment]
Thanks for your interest in our work! For the first two examples, you may try resampling or using dedicated checkpoints for outdoor and indoor respectively. For the last two examples, it is exactly the expected behavior. HDR map is not suitable for directly visualizing as LDR images as it is stored in the linear range. If you do so, you will get "over exposures" as the radiance above 255 is clipped. Please use tools like Blender or OpenHDR, where you can change camera exposures.

**Comment by caojiehui** [comment]
> Thanks for your interest in our work! For the first two examples, you may try resampling or using dedicated checkpoints for outdoor and indoor respectively. For the last two examples, it is exactly the expected behavior. HDR map is not suitable for directly visualizing as LDR images as it is stored in the linear range. If you do so, you will get "over exposures" as the radiance above 255 is clipped. Please use tools like Blender or OpenHDR, where you can change camera exposures.
As per your advice, after switching to the outdoor model, the generated results now significantly differ from the textual descriptions.

```
python text2light.py \
    -rg /data1/chc/T2L/model/Text2Light/global_sampler_clip \
    -rl /data1/chc/T2L/model/Text2Light/local_sampler_indoor \
    --outdir ./generated_panorama \
    --text "Elephant, Watering Hole, Baby Elephant" \
    --clip clip_emb.npy \
    --sritmo /data1/chc/T2L/model/Text2Light/sritmo.pth \
    --sr_factor 4
```
![holistic_ purple petal flower](https://github.com/FrozenBurning/Text2Light/assets/14066267/2ef3746b-925b-4047-a124-ede75e59b16e)

**Comment by caojiehui** [comment]
> Thanks for your interest in our work! For the first two examples, you may try resampling or using dedicated checkpoints for outdoor and indoor respectively. For the last two examples, it is exactly the expected behavior. HDR map is not suitable for directly visualizing as LDR images as it is stored in the linear range. If you do so, you will get "over exposures" as the radiance above 255 is clipped. Please use tools like Blender or OpenHDR, where you can change camera exposures.

As per your advice, after switching to the outdoor model, the generated results now significantly differ from the textual descriptions.

```
python text2light.py \
    -rg /data1/chc/T2L/model/Text2Light/global_sampler_clip \
    -rl /data1/chc/T2L/model/Text2Light/local_sampler_outdoor \
    --outdir ./generated_panorama \
    --text "purple petal flower" \
    --clip clip_emb.npy \
    --sritmo /data1/chc/T2L/model/Text2Light/sritmo.pth \
    --sr_factor 4
```
![holistic_ purple petal flower](https://github.com/FrozenBurning/Text2Light/assets/14066267/08b71fab-5dde-4d04-baca-03031ae38e00)

**Comment by FrozenBurning** [comment]
This text prompt is an OOD description and is not a scene-level context, of which we've discussed in the paper as limitations and future work. 

Close due to inactivity. Feel free to reopen for further questions!
